core/templates/components/code-mirror/codemirror-mergeview.component.ts

```diff
@@ -27,14 +28,24 @@ import { downgradeComponent } from '@angular/upgrade/static';
 export class CodemirrorMergeviewComponent implements
   AfterViewInit, OnInit, OnChanges {
   @Input() options = {};
-  @Input() leftValue;
-  @Input() rightValue;
-  codeMirrorInstance: CodeMirror.MergeView.MergeViewEditor;
-  constructor(private elementRef: ElementRef, private ngZone: NgZone) { }
+  // Below properties have been assigned as empty string if undefined.
```

Comment:
I am a bit confused. Is there actually a valid case where this component is allowed to be called with either leftValue or rightValue undefined? If so, what is it?

If not, then should we just set it to '' as default values and have them be string-only forever (and never undefined)?

Reply:
`leftValue` and `rightValue` are input properties, It is assigned after component communication.It can be the case sometimes that these properties aren't used ,one such instance is `logic-proof-interaction.component.html` where it is set as follows
```
    <oppia-codemirror (onLoad)="codeEditor($event)"
                      [readOnly]="!interactionIsActive">
    </oppia-codemirror>
```
So these would be undefined here and we would have to explicitly check for them to be not undefined

Reply:
Doesn't the logic proof interaction use the codemirror component rather than the codemirror mergeview? If so, it doesn't quite serve as an example of the problem, right?

Reply:
Ah sorry for this I didn't notice it carefully codemirror-mergeview has been used as a selector only at `state-diff-modal,controller,ts` where these properties have been passed.If this isn't the case I can refractor it for removing undefined but with confirmation of @srijanreddy98

Reply:
Yup, i am fine with a refactor.